ssl/quic/quic_impl.c

```diff
@@ -4729,7 +4729,9 @@ static QUIC_CONNECTION *create_qc_from_incoming_conn(QUIC_LISTENER *ql, QUIC_CHA
         goto err;
     }
 
-    ossl_quic_channel_get_peer_addr(ch, &qc->init_peer_addr); /* best effort */
+    if (csm_analyse_init_peer_addr(ossl_quic_port_get_net_wbio(ql->port), &qc->init_peer_addr))
+        ossl_quic_channel_set_peer_addr(ch, &qc->init_peer_addr);
```

Comment:
Not sure I understand the change here. We change from calling `ossl_quic_channel_get_peer_addr` which get the channel's `curr_peer_addr` if there is one. Instead we call `csm_analyse_init_peer_addr` which queries the underlying BIO and gets the BIO_ADDR from there if it is in a connected state. But `create_qc_from_incoming_conn` is a server side only function. So, usually we would not be using a connected socket?

Can you explain this more?

Reply:
Yep! Since we now propagate the peer in ch_on_new_conn_common() via ossl_quic_channel_set_peer_addr(ch, peer), the channel is already in "addressed" mode. This makes my change and also the original "best effort" call to ossl_quic_channel_get_peer_addr() at this place obsolete.  I’ve dropped it; leaving the original code would be just a source for confusion. All regression tests still pass.